services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentService.kt

```diff
@@ -102,4 +111,19 @@ class ApplicationAttachmentService(
             throw AttachmentUploadException("Infected file detected, see previous logs.")
         }
     }
+
+    /**
+     * Attachment should be sent if application is in Allu but has not yet proceeded to handling
+     * (status PENDING). If PENDING, application is in Allu and alluId should exist.
+     */
+    private fun sendIfPending(application: Application, attachment: ApplicationAttachmentEntity) =
+        with(application) {
+            logger.info {
+                "Check application if should send attachment, alluId: '$alluid', alluStatus: '$alluStatus'"
+            }
+            if (alluStatus == PENDING) {
```

Comment:
`PENDING_CLIENT` applications should be updated as well.

Having separate logic for whether an application should be updated in Allu is not great. `ApplicationService.isStillPending` would be a good place, but it updates the status from Allu, which is probably not necessary here.

Reply:
Actually, checking the status from Allu wouldn't be terrible here. I was thinking of a different scenario, but this is a separate endpoint call. I think Allu accepts attachments also when the application is in HANDLING, so the upload will succeed whether the application is pending or not. I think we should check the current status from Allu here, not rely on what the status was during the last background update.

Reply:
Maybe:
```suggestion
            if (alluId != null && isStillPending(application)) {
```

Reply:
So if HANDLING is acceptable here, we can't use isStillPending call.

Reply:
HANDLING shouldn't be allowed here, I meant Allu accepts attachments regardless of the status.

Reply:
Fixed.